web/src/components/AudienceContext.js

```diff
@@ -0,0 +1,22 @@
+import React, { createContext, useState } from "react";
+
+export const ageContext = createContext(null);
+
+export function AudienceContext({ children }) {
+  const [audience, setAudience] = useState(
+    localStorage.getItem("age") || localStorage.setItem("age", "5") || "5",
+  );
```

Comment:
Still a little uneasy about the readability of this construct, I think it would be clearer if we did an explicit useEffect, what do you think?

```suggestion
  const DEFAULT_AUDIENCE = 5
  const [audience, setAudience] = useState(DEFAULT_AUDIENCE);
  
  // when page loads, check local-storage for saved state
  useEffect(() => {
      const storedAudience = localStorage.getItem("age")
      if (!storedAudience) updateLocalStorage(DEFAULT_AUDIENCE)
      else setAudience(storedAudience)
  }, [])
```